woocommerce/woocommerce-ios
# [Unified Order Editing] Update New Order view and viewmodel to use existing order

Closes #6962.
Closes #7076.
~Closes~ Relates #6961.

## Description

This PR updates status New Order flow to accept, display and edit existing order.

### How

- Added initialOrder parameter to `NewOrderViewModel`, `RemoteOrderSynchronizer`.
- Updated `NewOrder` view to display data for existing order.
- Updated navigation in order details and new order to handle editing experience.

### Known issues

 Customer note and status won't be updated, it will require additional changes.

## Testing

1. Go to the Order tab, open any existing order.
2. Wait until order fully loads. Tap "•••" in navbar. Select "Edit".
3. Confirm that all relevant data is displayed on the form and no additional sync happened.
4. Try updating product list or customer data.
5. Close the editing flow with "Done" button or a swipe.
6. Confirm that Order Details shows updated order data.
7. Check that existing New Order flow is working without any issues.

## Screenshots

details|editing
--|--
![1](https://user-images.githubusercontent.com/3132438/172896680-3f236736-c209-4778-a02a-b2d97544549a.png)|![2](https://user-images.githubusercontent.com/3132438/172896906-c8097598-0bcf-4242-8e33-6f8f2ae73c76.png)

---
